prover/prover.go

```diff
@@ -461,25 +461,69 @@ func (p *Prover) onBlockProposed(
 			"proofWindow", block.ProofWindow,
 		)
 
-		proofWindowExpired := uint64(time.Now().Unix()) > block.ProposedAt+block.ProofWindow
-		// zero address means anyone can prove, proofWindowExpired means anyone can prove even if not zero address
-		if block.AssignedProver != p.proverAddress && block.AssignedProver != zeroAddress && !proofWindowExpired {
-			log.Info("Proposed block not proveable", "blockID", event.BlockId, "prover", block.AssignedProver.Hex())
+		var skipProofWindowExpiredCheck bool = false
 
-			// if we cant prove it
-			p.currentBlocksWaitingForProofWindowMutex.Lock()
-			p.currentBlocksWaitingForProofWindow = append(p.currentBlocksWaitingForProofWindow, event.Meta.Id)
-			p.currentBlocksWaitingForProofWindowMutex.Unlock()
+		if p.cfg.OracleProver {
```

Comment:
emmm looks like its in the event handler for BlockProposed? shall we move this to a handler for BlockProven which may make things simpler?